### PR TITLE
Quote Wise expenses in PayExpenseModal and preview Fees

### PR DESCRIPTION
**Pull request by kewitz**
Solves https://github.com/opencollective/opencollective/issues/3089.
Requires https://github.com/opencollective/opencollective-api/pull/6932.

https://www.loom.com/share/2e82b9c05f994fbe91058ec26bf37c12

### CODE REVIEW

File: components/expenses/PayExpenseModal.js [hunk 1]
```diff
@@ -123,6 +141,17 @@ const PayExpenseModal = ({ onClose, onSubmit, expense, collective, host, error }
...
+  const quoteQuery = useQuery(quoteExpenseQuery, {
+    variables: { id: expense.id },
+    context: API_V2_CONTEXT,
+    skip: !canQuote,
+  });
+
+  const paymentProcessorFee = quoteQuery?.data?.expense?.quote.paymentProcessorFeeAmount || {
+    valueInCents: formik.values.paymentProcessorFee,
+    currency: expense.currency,
+  };
```
Comment (Betree) [inline components/expenses/PayExpenseModal.js:154]: Won't this ignore the user input for payment processor fees when selecting "Manual" payment?

Comment (kewitz) [reply components/expenses/PayExpenseModal.js:154]: Oh, nice catch. Let me check and fix this.

**Comment by kewitz** [comment]
Rebased.

File: components/expenses/PayExpenseModal.js [hunk 2]
```diff
@@ -252,35 +284,48 @@ const PayExpenseModal = ({ onClose, onSubmit, expense, collective, host, error }
...
+            <AmountLine borderTop="0.8px dashed #9D9FA3">
+              <Label>
+                <FormattedMessage id="PayExpense.ProcessorFeesInput" defaultMessage="Payment processor fees" />
+              </Label>
+              <Amount>
+                <LoadingPlaceholder height="16px" />
+              </Amount>
+            </AmountLine>
+          )}
+          {canQuote && paymentProcessorFee?.valueInCents && (
```
Comment (Betree) [inline components/expenses/PayExpenseModal.js:297]: @kewitz I'm not sure about the `canQuote` part of this condition, don't we want to surface the fees whenever they're there? I ended up in this situation while testing and was surprised to not see the fees in the breakdown.

image

Comment (kewitz) [reply components/expenses/PayExpenseModal.js:297]: True. I'm fixing that.

**Pull request merged**